### ISSUE TITLE
How are dates handled

### ISSUE THREAD

**Issue by kmccurley** [IACR/latex#200]
I apologize in advance for the length of this issue, but I wanted to record the discussion about dates. We had a request from @yyyyx4 to show a date on the style:
> a small suggestion for the LaTeX template: Could it include a date by default? When doing literature research on 
> some topic, it's crucial to know if a paper is from 50 years ago or from last week, thus I think this should be written on 
> the document itself.

> (Presumably this should be a metadata setting for the paper, like the title and authors, so that spurious recompilations 
> don't suddenly make something seem much newer than it really is.)

This is certainly a good suggestion, but it raises the issue of how we handle dates associated with a paper, because there are several dates and different uses for the dates. First, there are several reasons to associate a date with the paper:
1. when I'm writing a paper, I like to have `\today` show up someplace so if I print a version I can later see which version I'm looking at. This means it's useful to have something in preprint mode (for the author).
2. when I'm reading a paper, it's nice to know when the author asserts that the paper is written. Of course, there is no reason to trust what the author wrote on the paper - they could have backdated it. This is why on the `final` version we show submission date and acceptance date, because those are the dates that are relevant for claiming provenance on the result. Unfortunately I later discovered that hotcrp has [no concept of "acceptance date"](https://github.com/IACR/latex-submit/issues/42) that is exposed, but the submission date is the upload date for the original submission and we save that. We can also capture the date when the final version is uploaded.
3. when a paper is cited as prior art related to a patent, you need to know when the idea was "made public". This may be unrelated to any dates asserted in the paper itself. Let's leave that problem to lawyers, but let's not have IACR make inappropriate assertions.

So long as we recognize the fact that dates supplied by the author are untrustworthy, I think it's reasonable to show a date on the `preprint` version of the paper (I guess we could have it on the `submission` version, but that should only be seen by the referees). The usual way of doing this is to use `\@date` in `\maketitle` to show the author's assertion of date. The author can set this by calling something like `\date{July 22, 1969}` but if they don't do that, it becomes the date when the PDF was produced (`\today`).

As for putting the dates into the PDF Info dictionary or XMP metadata, there are several problems with the standard schemas (and how `hyperxmp` uses them). `hyperxmp` has several keys that are used to set dates:
1. `pdfcreationdate` should really be the thing taken from the LaTeX compilation, namely `\today`. This becomes `xmp:CreateDate` in XMP. The official definition is:

> The date and time the resource was created. For a digital file, this need not
match a file-system creation time. For a freshly created resource, it should
be close to that time, modulo the time taken to write the file. Later file
transfer, copying, and so on, can make the file-system time arbitrarily
different.
Note that the documentation talks about when the PDF was created - not the LaTeX.
2. `pdfmoddate` seems to assume that PDFs would be updated after the LaTeX is compiled (perhaps to insert XMP). This becomes `xmp:ModifyDate` in XMP. The official definition is:

> The date and time the resource was last modified. 
This most naturally matches `\@today` because it talks about when the PDF was created (not the LaTeX).
3. `pdfmetadate` becomes `xmp:MetadataDate` in XMP. This is apparently [only used to give a timestamp](https://archimedespalimpsest.net/Documents/External/XMP/XMPSpecificationPart1.pdf) on the XMP package in case there are multiple XMP packages in the PDF file. The official definition is 

> The date and time that any metadata for this resource was last changed. It should be the same as or more recent than xmp:ModifyDate. 

None of these correspond to the dates that correspond to submission date or acceptance date on the LaTeX, and it's common for a PDF to contain multiple versions of them (if the PDF gets changed by Adobe tools or something else). Of course the XMP schema can be extended, but different publishers will extend it in different ways. Note that `hyperxmp` has no support for that. Springer inserts the following:
```
<crossmark:MajorVersionDate>2010-04-23</crossmark:MajorVersionDate>
<pdfx:CrossmarkMajorVersionDate>2010-04-23</pdfx:CrossmarkMajorVersionDate>
```
(the latter is a mirror of the former, so I'm not sure why they insert both). They define it as "The date when a publication was published." 

In an Elsevier paper to Journal of Number Theory, I found the following:
```
<prism:coverDisplayDate>April 2021</prism:coverDisplayDate>
<crossmark:MajorVersionDate>2010-04-23</crossmark:MajorVersionDate>
<pdfx:CrossmarkMajorVersionDate>2010-04-23</pdfx:CrossmarkMajorVersionDate>
```
This is fairly consistent with what Springer uses, and I found similar things in papers from the London Math Society, PLOS, and jmc.org. In an ACM paper I found:
```
<dc:date>
  <rdf:Seq>
      <rdf:li>2021-12-30T17:05:29Z</rdf:li>
   </rdf:Seq>
</dc:date>   
```
This is pretty useless because Dublin core does not define what these dates mean. ACM also has some other PDF stream objects with dates in them that don't seem to correspond to any standard. I think they are conference dates.

This makes me think we should be inserting `crossmark:MajorVersionDate` for the date of the final submission in our papers, but we have no mechanism to do that at present. We could build something in the publication pipeline using the `xmpincl` package, but `pdfx` and `hyperxmp` have no support for this.

To summarize:
1. we can print whatever we want on the PDF for preprint or submission versions. final versions already have submission and acceptance dates. Perhaps it is appropriate to modify `\@maketitle` to do this.
2. the XMP metadata produced by `hyperxmp` is insufficient. We should probably insert the crossmarkMajorVersionDate to indicate the date when the final version was uploaded. At this point the LaTeX tools for inserting XMP metadata are inadequate.

**Comment by jwbos** [comment]
Indeed, we shouldn't overcomplicate this. I like your point (1) in the summary. Final versions *should* have a date and this we already support. For preprint we should simply support an option for the author to provide a date (which they can choose) and if provided is shown in the footer on the frontpage. I can work on a proposal.

**Comment by jwbos** [comment]
First attempt here: #212
